### ISSUE TITLE
preg_match_all errors with router in Lumen v5.1.1

### ISSUE THREAD

**Issue by bviolier** [dingo/api#496]
Hi,

I just installed a fresh version of Lumen v5.1.1 with the dingo/api dev-master.

I followed the installation instructions and added to app.php:
````php 
$app->register(Dingo\Api\Provider\LumenServiceProvider::class);
$api = $app['api.router'];
````

To my routes.php I added:
````php
$api->version('v1', function ($api) {
    $api->get('number', 'App\Http\Controllers\v1\NumberController');
});
````

Whatever I fill in in the first or second parameter of the get (or if I do post), I keep getting the following error:
````
ErrorException in Std.php line 26:
preg_match_all(): Compilation failed: internal error: previously-checked referenced subpattern not found at offset 99

in Std.php line 26
at Application->Laravel\Lumen\{closure}('2', 'preg_match_all(): Compilation failed: internal error: previously-checked referenced subpattern not found at offset 99', '/var/www/service/current/vendor/nikic/fast-route/src/RouteParser/Std.php', '26', array('route' => '/api/number', 'matches' => null))
at preg_match_all('~\{ \s* ([a-zA-Z][a-zA-Z0-9_]*) \s* (?: : \s* ([^{}]*(?:\{(?-1)\}[^{}]*)*) )? \}~x', '/api/number', null, '258') in Std.php line 26
at Std->parse('/api/number') in RouteCollector.php line 30
````

I tried replacing the code in Std.php with the newest version of nikic/fast-route, but that doesn't help anything. I also tried using API_PREFIX or API_DOMAIN.

I run PHP v5.6.10-1, but before this I ran v5.5.29, and had the same issues.  

Am I doing something wrong here or is this a bug between specific versions?

PS. When I run the code just through PHP, it works.
````php
<?php
	
preg_match_all('~\{ \s* ([a-zA-Z][a-zA-Z0-9_]*) \s* (?: : \s* ([^{}]*(?:\{(?-1)\}[^{}]*)*) )? \}~x', '/api/number', $matches, '258');
var_dump($matches);
````

**Comment by jasonlewis** [comment]
Um, you're missing the the method for that route. You specify the controller by not the `@method` that the route uses.

```php
$api->get('number', 'App\Http\Controllers\v1\NumberController@index');
```

You'll also need to get the instance of the router in the `routes.php` file as it's no longer handed between the bootsrapping file and the routes file.

**Comment by bviolier** [comment]
Sorry for not clearing that up earlier. Not having the @method in the examples is just because I almost tried everything. Also with the @method it doesn't work. 
The router instance I currently get in the bootstrap/app.php, and that part seems to work. But I started with getting the instance in routes.php, and I switched it up to the app.php to see if that helped.

Also, when change Std.php to not use the preg_match_all, but just do as if the preg_match_all didn't have any results, it all works okay. So it really seems to be some word combination issue.
````php
public function parse($route) {
    return [$route];
}
````

note: i will edit my original post to make sure everything is clear.
